### PR TITLE
[Docs]Add Chinese translation of docs_zh-CN/tutorials/customize_dataset.md

### PR DESCRIPTION
**Pull request by justiceeem**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 1]
```diff
@@ -1 +1,459 @@
 # 教程 2: 自定义数据集
+
+## 支持新的数据格式
+
+为了支持新的数据格式，可以选择将数据转换成现成的格式（COCO或者PASCAL）或将其转换成中间格式。当然也可以选择以离线的形式（在训练之前使用脚本转换）或者在线的形式（实现一个新的dataset在训练中进行转换）来转换数据。
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:5]: 中文 Engilsh 中文；中文 数字 中文

中文和英文，中文和数字之间需要有空格
请检查全文，

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 2]
```diff
@@ -1 +1,459 @@
...
+
+在MMDetection中，建议将数据转换成COCO格式并以离线的方式进行，因此在完成数据转换后只需修改配置文件中的标注数据的路径和类别即可。
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:7]: space

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 3]
```diff
@@ -1 +1,459 @@
...
+
+### 调整新的数据格式为现有的数据格式
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:9]: `将新的数据格式转换为现有的数据格式` is better

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 4]
```diff
@@ -1 +1,459 @@
...
+
+最简单的方法就是将你的数据集转换成现有的数据格式（COCO或者PASCAL VOC）
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:11]: ...格式（COCO 或者 PASCAL VOC）

space

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 5]
```diff
@@ -1 +1,459 @@
...
+
+COCO格式的json标注文件有如下必要的键：
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:13]: space

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 6]
```diff
@@ -1 +1,459 @@
...
+    },
+    ...
+],
+
+'categories': [
+    {'id': 0, 'name': 'car'},
+ ]
+```
+
+在json文件中有三个必要的键：
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:48]: space

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 7]
```diff
@@ -1 +1,459 @@
...
+
+- `images`: 包含多个图片以及它们的信息的数组，比如 `file_name`，`height`，`width`，和 `id`。
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:50]: 例如 `file_name`、`height`、`width` 和 `id`。

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 8]
```diff
@@ -1 +1,459 @@
...
+- `annotations`: 包含多个实例标注信息的数组。
+- `categories`: 包含多个类别名字和ID的数组。
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:52]: space

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 9]
```diff
@@ -1 +1,459 @@
...
+
+在数据预处理之后，使用现成的数据格式来训练自定义的新数据集还有两步要做（以COCO为例）：
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:54]: - 现有的数据格式
- space

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 10]
```diff
@@ -1 +1,459 @@
...
+
+1. 为自定义数据集修改配置文件。
+2. 检查自定义数据集的标注。
+
+这里给出用包含5个类别的自定义数据集（COCO格式）训练现成的Cascade MaskRCNN R50 FPN检测器为例，来展示如上的两个步骤。
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:59]: 这里我们举一个例子来展示上面的两个步骤，这个例子使用包括 5 个类别的 COCO 格式的数据集来训练一个现有的 Cascade MaskRCNN R50 FPN 检测器
`有更好的翻译也可以，当前的翻译有点不通顺`

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 11]
```diff
@@ -1 +1,459 @@
...
+
+#### 1. 为自定义数据集修改配置文件
+
+配置文件的修改涉及两个方面：
+
+1. `data`部分。需要在`data.train`，`data.val`和`data.test`中添加`classes`。
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:65]: 1. `data` 部分。需要在 `data.train`、`data.val` 和 `data.test` 中添加 `classes`。

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 12]
```diff
@@ -1 +1,459 @@
...
+2. `model`部分中的`num_classes`。需要将默认值（COCO数据集中为80）修改为自定义数据集中的类别数。
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:66]: space

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 13]
```diff
@@ -1 +1,459 @@
...
+
+`configs/my_custom_config.py`内容如下：
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:68]: space

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 14]
```diff
@@ -1 +1,459 @@
...
+                type='Shared2FCBBoxHead',
+                # 将所有的`num_classes`默认值修改为5（原来为80）
+                num_classes=5)],
+    # 将所有的`num_classes`默认值修改为5（原来为80）
+    mask_head=dict(num_classes=5)))
+```
+
+#### 2. 检查自定义数据集的标注
+
+假设你自己的数据集是COCO格式，那么需要保证数据的标注没有问题：
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:124]: space

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 15]
```diff
@@ -1 +1,459 @@
...
+
+# 1. 数据集设定
+dataset_type = 'CocoDataset'
+classes = ('a', 'b', 'c', 'd', 'e')
+data = dict(
+    samples_per_gpu=2,
+    workers_per_gpu=2,
+    train=dict(
+        type=dataset_type,
+        # 将类别名字添加至`classes`字段中
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:83]: space

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 16]
```diff
@@ -1 +1,459 @@
...
+        classes=classes,
+        ann_file='path/to/your/train/annotation_data',
+        img_prefix='path/to/your/train/image_data'),
+    val=dict(
+        type=dataset_type,
+        # 将类别名字添加至`classes`字段中
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:89]: space

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 17]
```diff
@@ -1 +1,459 @@
...
+        classes=classes,
+        ann_file='path/to/your/val/annotation_data',
+        img_prefix='path/to/your/val/image_data'),
+    test=dict(
+        type=dataset_type,
+        # 将类别名字添加至`classes`字段中
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:95]: space

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 18]
```diff
@@ -1 +1,459 @@
...
+        classes=classes,
+        ann_file='path/to/your/test/annotation_data',
+        img_prefix='path/to/your/test/image_data'))
+
+# 2. 模型设置
+
+# 将所有的`num_classes`默认值修改为5（原来为80）
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:102]: space

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 19]
```diff
@@ -1 +1,459 @@
...
+model = dict(
+    roi_head=dict(
+        bbox_head=[
+            dict(
+                type='Shared2FCBBoxHead',
+                # 将所有的`num_classes`默认值修改为5（原来为80）
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:108]: space

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 20]
```diff
@@ -1 +1,459 @@
...
+                num_classes=5),
+            dict(
+                type='Shared2FCBBoxHead',
+                # 将所有的`num_classes`默认值修改为5（原来为80）
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:112]: space

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 21]
```diff
@@ -1 +1,459 @@
...
+                num_classes=5),
+            dict(
+                type='Shared2FCBBoxHead',
+                # 将所有的`num_classes`默认值修改为5（原来为80）
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:116]: space

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 22]
```diff
@@ -1 +1,459 @@
...
+                num_classes=5)],
+    # 将所有的`num_classes`默认值修改为5（原来为80）
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:118]: space

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 23]
```diff
@@ -1 +1,459 @@
...
+    mask_head=dict(num_classes=5)))
+```
+
+#### 2. 检查自定义数据集的标注
+
+假设你自己的数据集是COCO格式，那么需要保证数据的标注没有问题：
+
+1. 标注文件中`categories`的长度要与配置中的`classes`元组长度相匹配，它们都表示有几类。（如例子中有5个类别）
+2. 配置文件中`classes`字段应与标注文件里`categories`下的`name`有相同的元素且顺序一致。MMDetection会自动将`categories`中不连续的`id`映射成连续的索引，因此`categories`下的`name`的字符串顺序会影响标签的索引。同时，配置文件中的`classes`的字符串顺序也会影响到预测框可视化时的标签。
+3. `annotations`中的`category_id`必须是合法的值。比如所有`category_id`的值都应该属于`categories`中的`id`。
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:128]: - 合法 -> 有效
- check space in 1.2.3

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 24]
```diff
@@ -1 +1,459 @@
...
+
+下面标注的例子是合法的：
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:130]: 下面是一个有效标注的例子

有更合适的翻译也可

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 25]
```diff
@@ -1 +1,459 @@
...
+
+```python
+
+'annotations': [
+    {
+        'segmentation': [[192.81,
+            247.09,
+            ...
+            219.03,
+            249.06]],  # if you have mask labels
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:140]: translate to Chinese

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 26]
```diff
@@ -1 +1,459 @@
...
+    ...
+],
+
+# MMDetection会自动将`categories`中不连续的`id`映射成连续的索引。
+'categories': [
+    {'id': 1, 'name': 'a'}, {'id': 3, 'name': 'b'}, {'id': 4, 'name': 'c'}, {'id': 16, 'name': 'd'}, {'id': 17, 'name': 'e'},
+ ]
+```
+
+我们使用这种方式来支持CityScapes数据集。脚本在[cityscapes.py](https://github.com/open-mmlab/mmdetection/blob/master/tools/dataset_converters/cityscapes.py) 并且我们提供了微调的[configs](https://github.com/open-mmlab/mmdetection/blob/master/configs/cityscapes).
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:157]: space

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 27]
```diff
@@ -1 +1,459 @@
...
+
+**注意**
+
+1. 对于实例分割数据集, **MMDetection目前只支持评估COCO格式的mask AP**.
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:161]: space

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 28]
```diff
@@ -1 +1,459 @@
...
+2. 推荐训练之前进行离线转换，这样就可以继续使用`CocoDataset`且只需修改标注文件的路径以及训练的种类。
+
+### 调整新的数据格式为中间格式
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:164]: check space in this section

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 29]
```diff
@@ -1 +1,459 @@
...
+
+如果不想将标注格式转换为COCO或者PASCAL格式也是可行的。实际上，我们定义了一种简单的标注格式并且与所有现成的数据格式兼容，也能进行离线或者在线转换。
+
+数据集的标注是包含多个字典（dict）的列表，每个字典（dict）都与一张图片对应。测试时需要用到`filename` （相对路径）， `width`，`height`三个字段；训练时则额外需要`ann`。`ann`也是一个包含了至少两个字段的字典：
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:168]: - `ann` 也是至少包含了两个字段的字典：
- merge the next paragraph

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 30]
```diff
@@ -1 +1,459 @@
...
+            'labels': <np.ndarray, int64> (n, ),
+            'bboxes_ignore': <np.ndarray, float32> (k, 4),
+            'labels_ignore': <np.ndarray, int64> (k, ) （可选字段）
+        }
+    },
+    ...
+]
+```
+
+处理自定义数据集的两种方法。
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:191]: 有两种方法处理自定义数据

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 31]
```diff
@@ -1 +1,459 @@
...
+
+- 在线转换（online conversion）
+
+  You can write a new Dataset class inherited from `CustomDataset`, and overwrite two methods
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:195]: translate to Chinese

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 32]
```diff
@@ -1 +1,459 @@
...
+  可以新写一个继承自`CustomDataset`的Dataset类，并重写`load_annotations(self, ann_file)`以及`get_ann_info(self, idx)`这两个方法，正如[CocoDataset](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/datasets/coco.py)与[VOCDataset](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/datasets/voc.py).
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:196]: 正如 -> 例如

**Comment by BIGWangYuDong** [comment]
Hi @justiceeem,
Thanks forr your kind contribution. Please re-request review if you fix my comments.

**Comment by BIGWangYuDong** [comment]
If you have already resolved the problem, you can click the  `Resolve conversation`

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 33]
```diff
@@ -1 +1,457 @@
 # 教程 2: 自定义数据集
+
+## 支持新的数据格式
+
+为了支持新的数据格式，可以选择将数据转换成现成的格式（ COCO 或者PASCAL ）或将其转换成中间格式。当然也可以选择以离线的形式（在训练之前使用脚本转换）或者在线的形式（实现一个新的 dataset 在训练中进行转换）来转换数据。
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:5]: （COCO 或者 PASCAL VOC）

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 34]
```diff
@@ -1 +1,457 @@
...
+1. 为自定义数据集修改配置文件。
+2. 检查自定义数据集的标注。
+
+这里我们举一个例子来展示上面的两个步骤，这个例子使用包括 5 个类别的 COCO 格式的数据集来训练一个现有的 Cascade MaskRCNN R50 FPN 检测器
+
+#### 1. 为自定义数据集修改配置文件
+
+配置文件的修改涉及两个方面：
+
+1. `data`部分。需要在`data.train`、`data.val`和`data.test`中添加`classes`。
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:65]: `data` 部分。需要在 `data.train`、`data.val` 和 `data.test` 中添加 `classes`。

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 35]
```diff
@@ -1 +1,457 @@
...
+2. `model`部分中的`num_classes`。需要将默认值（ COCO 数据集中为 80 ）修改为自定义数据集中的类别数。
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:66]: `model` 部分中的 `num_classes`。需要将默认值（COCO 数据集中为 80）修改为自定义数据集中的类别数。

File: docs_zh-CN/tutorials/customize_dataset.md (hunk 33 above)
Comment (BIGWangYuDong) [reply docs_zh-CN/tutorials/customize_dataset.md:5]: - 中文标点和英文之间不需要加空格

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 36]
```diff
@@ -1 +1,457 @@
...
+
+## 支持新的数据格式
+
+为了支持新的数据格式，可以选择将数据转换成现成的格式（ COCO 或者PASCAL ）或将其转换成中间格式。当然也可以选择以离线的形式（在训练之前使用脚本转换）或者在线的形式（实现一个新的 dataset 在训练中进行转换）来转换数据。
+
+在 MMDetection 中，建议将数据转换成 COCO 格式并以离线的方式进行，因此在完成数据转换后只需修改配置文件中的标注数据的路径和类别即可。
+
+### 将新的数据格式转换为现有的数据格式
+
+最简单的方法就是将你的数据集转换成现有的数据格式（ COCO 或者 PASCAL VOC）
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:11]: （COCO 或者 PASCAL VOC）

**Review by BIGWangYuDong — COMMENTED** [review]
Please resolve my comments one by one. Thx

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 37]
```diff
@@ -1 +1,457 @@
...
+
+为了支持新的数据格式，可以选择将数据转换成现成的格式（COCO 或者PASCAL）或将其转换成中间格式。当然也可以选择以离线的形式（在训练之前使用脚本转换）或者在线的形式（实现一个新的 dataset 在训练中进行转换）来转换数据。
+
+在 MMDetection 中，建议将数据转换成 COCO 格式并以离线的方式进行，因此在完成数据转换后只需修改配置文件中的标注数据的路径和类别即可。
+
+### 将新的数据格式转换为现有的数据格式
+
+最简单的方法就是将你的数据集转换成现有的数据格式（COCO 或者 PASCAL VOC）
+
+COCO 格式的 json 标注文件有如下必要的键：
```
Comment (ZwwWayne) [inline docs_zh-CN/tutorials/customize_dataset.md:13]: 键->字段 might be better

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 38]
```diff
@@ -1 +1,457 @@
...
+ ]
+```
+
+在 json 文件中有三个必要的键：
+
+- `images`: 包含多个图片以及它们的信息的数组，例如 `file_name`、`height`、`width` 和 `id`。
+- `annotations`: 包含多个实例标注信息的数组。
+- `categories`: 包含多个类别名字和 ID 的数组。
+
+在数据预处理之后，使用现有的数据格式来训练自定义的新数据集还有两步要做（以 COCO 为例）：
```
Comment (ZwwWayne) [inline docs_zh-CN/tutorials/customize_dataset.md:54]: 还有两步要做->有如下两步

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 39]
```diff
@@ -1 +1,457 @@
...
+
+1. 为自定义数据集修改配置文件。
+2. 检查自定义数据集的标注。
+
+这里我们举一个例子来展示上面的两个步骤，这个例子使用包括 5 个类别的 COCO 格式的数据集来训练一个现有的 Cascade MaskRCNN R50 FPN 检测器
```
Comment (ZwwWayne) [inline docs_zh-CN/tutorials/customize_dataset.md:59]: Cascade Mask R-CNN R50-FPN

File: docs_zh-CN/tutorials/customize_dataset.md (hunk 15 above)
Comment (BIGWangYuDong) [reply docs_zh-CN/tutorials/customize_dataset.md:83]: 将类别名字添加至 `classes` 字段中

File: docs_zh-CN/tutorials/customize_dataset.md (hunk 17 above)
Comment (BIGWangYuDong) [reply docs_zh-CN/tutorials/customize_dataset.md:95]: 将类别名字添加至`classes`字段中

File: docs_zh-CN/tutorials/customize_dataset.md (hunk 18 above)
Comment (BIGWangYuDong) [reply docs_zh-CN/tutorials/customize_dataset.md:102]: 将所有的 `num_classes` 默认值修改为 5（原来为 80）

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 40]
```diff
@@ -1 +1,457 @@
...
+**注意**
+
+1. 对于实例分割数据集, **MMDetection 目前只支持评估 COCO 格式的 mask AP**.
+2. 推荐训练之前进行离线转换，这样就可以继续使用 `CocoDataset` 且只需修改标注文件的路径以及训练的种类。
+
+### 调整新的数据格式为中间格式
+
+如果不想将标注格式转换为 COCO 或者 PASCAL 格式也是可行的。实际上，我们定义了一种简单的标注格式并且与所有现有的数据格式兼容，也能进行离线或者在线转换。
+
+数据集的标注是包含多个字典（ dict ）的列表，每个字典（ dict ）都与一张图片对应。测试时需要用到 `filename` （相对路径）， `width`，`height`三个字段；训练时则额外需要 `ann`。`ann` 也是至少包含了两个字段的字典：`bboxes` 和 `labels`，它们都是 numpy array。有些数据集可能会提供如：crowd/difficult/ignored bboxes 标注，那么我们使用 `bboxes_ignore` 以及 `labels_ignore` 来包含它们。
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:168]: - 字典（dict） * 2
- 测试时需要用到 `filename`（相对路径）、`width` 和 `height` 三个字段   `delete the space`
-

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 41]
```diff
@@ -1 +1,457 @@
...
+classes = 'path/to/classes.txt'
+data = dict(
+    train=dict(classes=classes),
+    val=dict(classes=classes),
+    test=dict(classes=classes))
+```
+
+**注意**
+
+- 在 MMDetection v2.5.0 之前，如果类别为集合时数据集将自动过滤掉空真实框的图片，且没办法通过设置配置将其关闭。这是一种不和预期的行为而且会引起混淆，因为当类别不是集合时数据集只有在 `filter_empty_gt=True` 以及 `test_mode=False` 的情况下才会过滤掉空真实框的图片。在 MMDetection v2.5.0 之后，我们将图片的过滤以及类别的修改进行解耦，如，数据集只有在 `filter_empty_gt=True` 和 `test_mode=False` 的情况下才会过滤掉空真实框的图片，无论类别是否为集合。设置类别只会影响用于训练的标注类别，用户可以自己决定是否过滤空真实框的图片。
```
Comment (ZwwWayne) [inline docs_zh-CN/tutorials/customize_dataset.md:454]: need to rephrase this line, eg., 不和预期，空真空框的图片，设置配置

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 42]
```diff
@@ -1 +1,457 @@
 # 教程 2: 自定义数据集
+
+## 支持新的数据格式
+
+为了支持新的数据格式，可以选择将数据转换成现成的格式（COCO 或者PASCAL）或将其转换成中间格式。当然也可以选择以离线的形式（在训练之前使用脚本转换）或者在线的形式（实现一个新的 dataset 在训练中进行转换）来转换数据。
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:5]: COCO 或者PASCAL -> COCO 或者 PASCAL

File: docs_zh-CN/tutorials/customize_dataset.md [hunk 43]
```diff
@@ -1 +1,457 @@
...
+data = dict(
+    train=dict(classes=classes),
+    val=dict(classes=classes),
+    test=dict(classes=classes))
+```
+
+**注意**
+
+- 在 MMDetection v2.5.0 之前，如果类别为集合时数据集将自动过滤掉空真实框的图片，且没办法通过设置配置将其关闭。这是一种不和预期的行为而且会引起混淆，因为当类别不是集合时数据集只有在 `filter_empty_gt=True` 以及 `test_mode=False` 的情况下才会过滤掉空真实框的图片。在 MMDetection v2.5.0 之后，我们将图片的过滤以及类别的修改进行解耦，如，数据集只有在 `filter_empty_gt=True` 和 `test_mode=False` 的情况下才会过滤掉空真实框的图片，无论类别是否为集合。设置类别只会影响用于训练的标注类别，用户可以自己决定是否过滤空真实框的图片。
+- 因为中间格式只有框的标签并不包含类别的名字，所以使用 `CustomDataset` 时用户不能通过设置配置来过滤掉空真实框的图片。但是可以通过离线的方式来解决。
```
Comment (BIGWangYuDong) [inline docs_zh-CN/tutorials/customize_dataset.md:455]: need to rephrase this line, eg., 过滤掉空真实框的图片

**Review by BIGWangYuDong — APPROVED** [review]

**Review by hhaAndroid — APPROVED** [review]

**Comment by OpenMMLab-Coodinator** [comment]
Hi @justiceeem ！First of all, we want to express our gratitude for your significant PR in this project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.

We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/raweFPmdzG

If you are Chinese or have WeChat，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤

**Pull request merged**